jmespath-jackson-jr/src/main/java/io/burt/jmespath/jacksonjr/JacksonJrRuntime.java

```diff
@@ -0,0 +1,225 @@
+
+            while (iterator.hasNext()) {
+                Map.Entry<String, TreeNode> entry = (Map.Entry) iterator.next();
+                list.add(entry.getValue());
+            }
+            return list;
+        } else {
+            return Collections.emptyList();
+        }
+    }
+
+    @Override
+    public String toString(TreeNode value) {
+        if (value.asToken().equals(JsonToken.VALUE_STRING)) {
+            return ((JrsString) value).asText();
+        } else {
+            try {
+                return json.asString(value);
+            } catch (IOException e) {
+                return "";
+            }
+        }
+    }
+
+    @Override
+    public Number toNumber(TreeNode value) {
+        if (value.isValueNode() && ((JrsValue) value).isNumber()) {
+            JrsNumber number = (JrsNumber) value;
+            return number.getValue();
+        } else return null;
+    }
+
+    @Override
+    public boolean isTruthy(TreeNode value) {
+        // false, null, empty lists, empty objects, empty strings.
+        if (value.isContainerNode()) {
+            return value.size() > 0;
+        } else if (value.isValueNode()) {
+            if (value.asToken().equals(JsonToken.VALUE_STRING)) {
+                return !((JrsString) value).asText().isEmpty();
+            } else return !value.asToken().equals(JsonToken.VALUE_FALSE) &&
+                    !value.asToken().equals(JsonToken.VALUE_NULL);
+        } else {
+            return !value.isMissingNode();
+        }
+    }
+
+    @Override
+    public JmesPathType typeOf(TreeNode value) {
+        switch (value.asToken()) {
+            case START_ARRAY:
+            case END_ARRAY:
+                return JmesPathType.ARRAY;
+            case VALUE_EMBEDDED_OBJECT:
+            case START_OBJECT:
+            case END_OBJECT:
+                return JmesPathType.OBJECT;
+            case VALUE_STRING:
+                return JmesPathType.STRING;
+            case VALUE_NUMBER_INT:
+            case VALUE_NUMBER_FLOAT:
+                return JmesPathType.NUMBER;
+            case VALUE_TRUE:
+            case VALUE_FALSE:
+                return JmesPathType.BOOLEAN;
+            case VALUE_NULL:
+                return JmesPathType.NULL;
+            case NOT_AVAILABLE:
+            default:
+               throw new IllegalStateException(String.format("Unknown node type encountered: %s", value.asToken()));
+        }
+    }
+
+    @Override
+    public TreeNode getProperty(TreeNode value, TreeNode name) {
+        if (value == null || value.asToken().equals(JsonToken.VALUE_NULL)) {
+            return JrsNull.instance();
+        } else {
+            TreeNode node = value.get(((JrsString) name).asText());
+            return node != null ? node : createNull();
+        }
+    }
+
+    @Override
+    public Collection<TreeNode> getPropertyNames(TreeNode value) {
+        if (value != null && value.isObject()) {
```

Comment:
Nitpick: I don't believe `value` can be `null` here. The existing Jackson implementation does not check for null. Unless I remember incorrectly the runtime would not pass a Java `null` to this method, only `TreeNode`s representing null (i.e. objects created with `#createNull`).

This comment also goes for `#getProperty` that also checks for null.

Reply:
I had some tests failing because of value being `null`

Reply:
fixed... it's not required